pkg/apis/infra/v1alpha1/cluster_types.go

```diff
@@ -0,0 +1,242 @@
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ClusterSpec   `json:"spec,omitempty"`
+	Status            ClusterStatus `json:"status,omitempty"`
+}
+
+// ClusterSpect defines the desired state of Cluster
+type ClusterSpec struct {
+	// Type is the type of the cluster.
+	Type ClusterProviderType `json:"type"`
+	// Credential is the credential to use to access the cloud provider.
+	// +optional
+	Credential *CredentialConfig `json:"credential,omitempty"`
+	// Version is the Kubernetes version to use for the cluster.
+	Version string `json:"version"`
+	// Region is the region to use for the cluster.
+	Region string `json:"region"`
+	// Network is the network configuration for the cluster.
+	Network NetworkConfig `json:"network"`
+	// Master is the configuration for the master node.
+	Master MasterConfig `json:"master"`
+	// Workers is the list of worker nodes.
+	Workers []WorkerConfig `json:"workers"`
+	// PodIdentity is the configuration for the pod identity.
+	// +optional
+	PodIdentity PodIdentityConfig `json:"podIdentity,omitempty"`
+	// AdditionalResources provides a way to automatically apply a set of resouces to cluster after it's ready.
+	// Note: the resouces will only apply once.
+	// +optional
+	AdditionalResources []ResourceRef `json:"additionalResources,omitempty"`
+}
+
+type ClusterProviderType string
+
+const (
+	// AWSClusterProviderType is the type for the AWS provider.
+	AWSClusterProviderType ClusterProviderType = "aws"
+)
+
+type CredentialConfig struct {
+	SecretRef string `json:"secretRef"`
+}
+
+type NetworkConfig struct {
+	// VPC is the configuration for the VPC.
+	VPC VPCConfig `json:"vpc"`
+	// PodCIDR is the CIDR block for pods in this cluster.
+	// Defaults to 192.168.0.0/16.
+	// +optional
+	PodCIDR *string `json:"podCIDR,omitempty"`
+	// ServiceCIDR is the CIDR block for services in this cluster.
+	// Defaults to 10.96.0.0/12.
+	// +optional
+	ServiceCIDR *string `json:"serviceCIDR,omitempty"`
+	// CNI is the configuration for the CNI.
+	CNI CNIConfig `json:"cni"`
+}
+
+type VPCConfig struct {
+	// ID defines a unique identifier to reference this resource.
+	// +optional
+	ID string `json:"id"`
+	// Name is the name of the VPC.
+	// if not set, the name will be generated from cluster name.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// CidrBlock is the CIDR block to be used when the provider creates a managed VPC.
+	// Defaults to 10.0.0.0/16.
+	// +optional
+	CidrBlock *string `json:"cidrBlock"`
+}
+
+type CNIConfig struct {
+	// Cilium is the configuration for the Cilium CNI.
+	// +optional
+	Cilium *CiliumConfig `json:"cilium,omitempty"`
+}
+
+type CiliumConfig struct {
+}
+
+type MasterConfig struct {
+	NodeConfig `json:",inline"`
+}
+
+type WorkerConfig struct {
+	NodeConfig `json:",inline"`
+	// Strategy to use to replace existing nodes with new ones.
+	// +optional
+	Strategy *WorkerNodeStrategy `json:"strategy,omitempty"`
```

Comment:
upgradeStragegy

Reply:
`WorkerNodeStrategy`, since it is in Worker we donot need WorklerPrefix anymore.

How about calling NodeUpgradeStrategy